File: Chapter 05/Python/section_5_4_server_monitor/after/infrastructure/adapters/console_adapter.py
from ...core.ports.alert_port import AlertPort

class ConsoleAdapter(AlertPort):
    """ADAPTER 2: The 'Dev' Adapter.
    
    This adapter provides immediate visual feedback during local development. 
    It proves that the Core remains identical regardless of whether the 
    alert goes to a cloud service or a terminal screen.
    """

    def send_alert(self, message: str) -> None:
        """Prints a high-visibility alert to the local console.
        
        Args:
            message (str): The alert message provided by the Core.
        """
        # ANSI escape codes for red text mimic professional UI feedback
        ansi_red = "\033[91m"
        ansi_reset = "\033[0m"
        
        # We wrap the output to ensure the 'Dev' context is clear, including ANSI codes for visual emphasis
        print(f"{ansi_red}(DEV ADAPTER) ALERT: {message}{ansi_reset}")

File: Chapter 05/JS/section_5_4_server_monitor/after/infrastructure/adapters/consoleAdapter.js
const AlertPort = require('../../core/ports/alertPort');

/**
 * ADAPTER 2: The "Dev" Adapter.
 * Proves that the Core doesn't care if the alert goes to a 
 * cloud messaging service or simply prints to the local screen.
 */
class ConsoleAdapter extends AlertPort {
    sendAlert(message) {
        // We use ANSI escape codes to mimic a real red alert,
        // but the Core logic remains completely unaware of this UI detail.
        const ANSI_RED = "\x1b[31m";
        const ANSI_RESET = "\x1b[0m";
        console.log(`${ANSI_RED}(DEV ADAPTER) ALERT: ${message}${ANSI_RESET}`);
    }
}

module.exports = ConsoleAdapter;